src/catalog/tablet_catalog.cc

```diff
@@ -150,10 +147,17 @@ const ::hybridse::codec::Row TabletTableHandler::Get(int32_t pos) {
 
 ::hybridse::codec::RowIterator* TabletTableHandler::GetRawIterator() {
     auto tables = std::atomic_load_explicit(&tables_, std::memory_order_acquire);
-    if (!tables->empty()) {
-        return new catalog::FullTableIterator(tables);
+    std::map<uint32_t, std::shared_ptr<openmldb::client::TabletClient>> tablet_clients;
+    for (uint32_t pid = 0; pid < partition_num_; pid++) {
+        if (tables->count(pid) == 0) {
+            auto accessor = table_client_manager_->GetTablet(pid);
+            if (accessor) {
+                tablet_clients.emplace(pid, accessor->GetClient());
+            }
+        }
     }
-    return nullptr;
+    DLOG(INFO) << "table size " << tables->size() << " tablet_clients size " << tablet_clients.size();
+    return new catalog::FullTableIterator(GetTid(), tables, tablet_clients);
```

Comment:
here it is a bit dangerous as previously it is assumed to be local tables only. `select *` may return duplicate results.

have to check

Reply:
there is a test case to cover this scene
https://github.com/4paradigm/OpenMLDB/blob/e9acd54366616e9e7211dd17db7dd01bbf07bbd2/src/catalog/distribute_iterator_test.cc#L98-L123

Reply:
so `FullTableIterator` is never used before in the sql engine?

Reply:
I mean, in the cluster mode, to handle the `select  * from t1`:
- send the query to all the tablets
- each tablet uses the `FullTableIterator` to traverse the `local` table (before this PR)
- collect the response from all tablets and merge the results

However, if changing the `FullTableIterator` to iterating over a distributed table (i.e., collecting both local and remote partitions), the result from each `tablet` will be a full table. After merging, results would be duplicated.

Reply:
`FullTableIterator` will make sure that there will not duplicated data, which will hold the  partition leader located in local memory and remote tablet client. 
`SQLCusterRouter` will send request to all tablets and merge response data in local previously. In this pr,  `SQLCusterRouter` will send only one tablet that main table located and iterator all data by `FullTableIterator`

Reply:
oh. overlook the changes in `SQLClusterRouter`